bench_utils/src/mkbench.rs

```diff
@@ -0,0 +1,1006 @@
+// Copyright © 2019-2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Benchmark harness runner to instantiate and configure benchmarks.
+//!
+//! The file exports two items:
+//!  - baseline_comparison: A generic function to compare a data-structure with
+//!    and without a log.
+//! - `ScaleBenchBuilder`: A struct that helps to configure criterion to
+//!    evaluate the scalability of a data-structure with node-replication.
+
+use std::collections::HashMap;
+use std::fmt::{self};
+use std::fs::OpenOptions;
+use std::hint::black_box;
+use std::marker::{PhantomData, Send, Sync};
+use std::num::NonZeroUsize;
+use std::path::Path;
+use std::sync::{Arc, Barrier};
+use std::thread::{self, JoinHandle};
+use std::time::{Duration, Instant};
+
+use csv::WriterBuilder;
+use log::*;
+
+use node_replication::nr::{AffinityChange, Dispatch, NodeReplicated, ReplicaId, ThreadToken};
+use rand::seq::SliceRandom;
+use rand::SeedableRng;
+use serde::Serialize;
+
+pub use crate::topology::ThreadMapping;
+use crate::{benchmark::*, topology::*, Operation};
+
+fn chg_affinity(af: AffinityChange) -> usize {
```

Comment:
Maybe we can add OS or Arch cfg for this function?

Reply:
Yes my plan was to have a default for Linux I will add this once I finished to add the liveness changes to CNR.